### PR TITLE
Fix(iOS) TimePicker bug selecting new Time

### LINKED ISSUE #4611: [iOS] Unable to select a new time on TimePicker (iOS 14 new Style)
## Current behavior
When using the TimePicker with the new iOS14 Style the keyboard is preventing from entering a value.

## Expected behavior
Users are capable of setting a new time on the TimePicker while using the iOS14 style

## How to reproduce it (as minimally and precisely as possible)
Create an iOS app and use the `TimePicker`. Set the `FeatureConfiguration.TimePicker.UseLegacyStyle = false` in the App.xaml.cs

## Workaround
Setting the `FeatureConfiguration.TimePicker.UseLegacyStyle = true` will use the previous style `TimePicker`

## Environment

Nuget Package:

- [x] Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia
- [ ] Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia
- [ ] Uno.SourceGenerationTasks
- [ ] Uno.UI.RemoteControl / Uno.WinUI.RemoteControl
- [ ] Other: 

Nuget Package Version(s):

Affected platform(s):

- [x] iOS
- [ ] Android
- [ ] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [ ] macOS
- [ ] Skia
  - [ ] WPF
  - [ ] GTK (Linux)
  - [ ] Tizen
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

IDE:

- [ ] Visual Studio 2017 (version: )
- [x] Visual Studio 2019 (version: )
- [x] Visual Studio for Mac (version: )
- [ ] Rider Windows (version: )
- [x] Rider macOS (version: )
- [ ] Visual Studio Code (version: )

Relevant plugins:

- [ ] Resharper (version: )

## Anything else we need to know?

time_picker_ios14

### PR DESCRIPTION
**Pull request by ajpinedam**
GitHub Issue (If applicable): #
closes #4611 

## PR Type

What kind of change does this PR introduce?

- Bugfix

## What is the current behavior?

Issues on TimePicker with iOS14+ style not allowing to select a new time because the keyboard was dismissing the Picker Dialog.

## What is the new behavior?
Preventing the keyboard from showing allows the users to pick the new Time "scrolling".

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Review by jeromelaban — APPROVED** [review]

File: src/Uno.UI/UI/Xaml/Controls/TimePicker/TimePickerSelector.iOS.cs [hunk 1]
```diff
@@ -42,7 +42,8 @@ private protected override void OnLoaded()
 			SetPickerTime(Time.RoundToNextMinuteInterval(MinuteIncrement));
 			SaveInitialTime();
 			_picker.ValueChanged += OnValueChanged;
-			
+			_picker.EditingDidBegin += OnEditingDidBegin;
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/Controls/TimePicker/TimePickerSelector.iOS.cs:45]: Should we avoid registering if `UseLegacyStyle` is `true` ?

Comment (ajpinedam) [reply src/Uno.UI/UI/Xaml/Controls/TimePicker/TimePickerSelector.iOS.cs:45]: Makes no difference. The Legacy Picker is a Wheel that does not trigger the keyboard

**Comment by jeromelaban** [comment]
@mergifyio backport release/stable/3.1

**Review by feartDev — COMMENTED** [review]
Approval

**Pull request merged**